clients/app/src/views/screens/CreateFanClub/screens/CreateFanClub/components/ChoosePhotoComponent.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import AttachImage, { IImageItem } from 'views/components/compositions/AttachImage';
+import { buildUltrasS3SDK } from 'stores/sdkBuilder';
+import { fileToBlob } from 'utils/helpers/fileToBlob';
+import { fanClubsStore } from '../../../store';
+import { IChoosePhotoComponentProps } from '../types';
+
+const sdk = buildUltrasS3SDK();
+
+const ChoosePhotoComponent: React.FC<IChoosePhotoComponentProps> = ({
+  title,
+  rounded,
+  field,
+  folder,
+}) => {
+  const { add: storeAdd } = fanClubsStore.useSelector('add');
+  const [uploading, setUploading] = React.useState<boolean>(false);
+
+  const updateImage = React.useCallback(
+    async (newImage: string | null) => {
+      // @TODO: remove old image from storage.
+      storeAdd.setFieldValue(field, newImage || '');
+    },
+    [field, storeAdd]
+  );
+
+  const onChoose = React.useCallback(
+    async (images: Array<IImageItem>) => {
```

Comment:
why we need this **async**s?

Reply:
because the upload function is called via `await`, we need to wait for the image upload to handle the saved URL on AWS.
look at lines 39 and 40 :)